src/WebJobs.Script/Workers/Rpc/FunctionRegistration/RpcFunctionInvocationDispatcher.cs

```diff
@@ -142,13 +139,18 @@ internal async void ShutdownWebhostLanguageWorkerChannels()
             await _webHostLanguageWorkerChannelManager?.ShutdownChannelsAsync();
         }
 
-        private void StartWorkerProcesses(int startIndex, Action startAction)
+        private void StartWorkerProcesses(int startIndex, Func<Task> startAction)
         {
-            for (var count = startIndex; count < _maxProcessCount; count++)
+            // Starting the worker processes in a different thread to avoid delaying in the host initialization
```

Comment:
This task is unobserved. Please add a test inducing a failure in the code executed in the lambda and ensure that is properly handled. As-is, potential exceptions here will go unhandled in the background.

Reply:
nit: also update the comment as this is just scheduling a new `Task`.

Reply:
FYI - Failure to start worker process is handled here: https://github.com/Azure/azure-functions-host/blob/dev/src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs#L136-L147 - we have test coverage for this

Reply:
Can you clarify the comment above? Is that additional information to share that a class of errors is properly handled and surfaced, or are you suggesting we leave this as-is?

The code you've pointed to handles a number of known potential exceptions, but it does not make what we have here safe. There are a of calls up to that point that could introduce a failure that would end up being thrown in the background, and that's if we only think about the current state of the code. As some of those code paths evolve in the future, chances of exposing the fragility here just keep growing.

Any concerns with being defensive here?

Reply:
Sorry. I should have provided a complete response. The code I referred to above is an example of how to induce worker start up errors. Yes, if there is a way to log Task errors - we should

Reply:
Added the logging and a test.